File: Data Structures & Algorithms/binary-search/submission-2.java
class Solution {
    public int search(int[] nums, int target) 
    {
        // Fix 1: Change nums.length to nums.length - 1
        return h(nums, 0, nums.length - 1, target);
    }
    
    public int h(int[] nums, int l, int h, int t)
    {
        if (l > h)
        {
            return -1;
        }
        
        int mid = l + (h - l) / 2; 
        
        if (nums[mid] == t)
        {
            return mid;
        }
        else if (nums[mid] > t)
        {
            h = mid - 1;
        }
        else
        {
            l = mid + 1;
        }
        return h(nums, l, h, t);
    }
}

File: Data Structures & Algorithms/is-subsequence/submission-2.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        return self.h(s,t,0,0)
    def h(self,s,t,m,n):
        if(m==len(s)):
            return True
        if(n==len(t)):
            return False
        if(s[m]==t[n]):
            return self.h(s,t,m+1,n+1)
        return self.h(s,t,m,n+1)
        


        